core/discovery/reducer/in.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package reducer
+
+import (
+	"github.com/mysteriumnetwork/node/market"
+)
+
+// In matches proposal if field value exists in array
+func In(field FieldSelector, valuesExpected ...interface{}) func(market.ServiceProposal) bool {
+	if len(valuesExpected) == 0 {
+		return All()
+	}
+
+	return Field(field, func(value interface{}) bool {
+		for _, valueExpected := range valuesExpected {
+			if value == valueExpected {
+				return true
+			}
+		}
+		return false
+	})
+}
+
+// InString matches proposal if string value exists in array
+func InString(field FieldSelector, valuesExpected ...string) func(market.ServiceProposal) bool {
```

Comment:
`In` function already receiving `valuesExpected ...interface{}`, we can pass string and ints with it. No need to have extra functions for `int` and `string`.

Reply:
No, no generics in Golang. This is why I created this PR - because noticed bug in live

Reply:
If this is a bug, we can have a test that reproduces it. 
Because now I can change the tests to `In` only, and it keeps working:
![image](https://user-images.githubusercontent.com/8612618/58864995-8fca4800-86d7-11e9-9c80-3860fc98e072.png)

Reply:
It's impossible then I have array `providerIDs []string` -> `In(fieldProviderID, providerIDs...)`